### PR TITLE
Add the tab[s] shortcodes.

### PR DESCRIPTION
**Pull request by mattmoor**
I found out about these here and they're exactly what I've been missing in my life: https://kubernetes.io/docs/contribute/style/hugo-shortcodes/#tabs

/assign @samodell @evankanderson @RichieEscarez 
/hold

I don't yet know how to validate or try this, so I'm sort of hoping a YOLO CI job might show something 🤞

### CODE REVIEW

**Comment by mattmoor** [comment]
Alright, this is no longer strictly based on K8s, and I was able to set this working.  See the screenshot in slack :D

File: layouts/shortcodes/tab.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{{ if .Parent }}
+	{{ $name := trim (.Get "name") " " }}
+	{{ if not (.Parent.Scratch.Get "tabs") }}
+	{{ .Parent.Scratch.Set "tabs" slice }}
+	{{ end }}
+	{{ with .Inner }}
+	{{ $.Parent.Scratch.Add "tabs" (dict "name" $name "content" . ) }}
+	{{ else }}
+	asdf
```
Comment (mattmoor) [inline layouts/shortcodes/tab.html]: Honestly, I don't understand enough about what's even going on here to know why there is an else/end 😅 

I was mostly pruning this down to the minimal file that worked for me.

**Comment by samodell** [comment]
@mattmoor  Can you drop the syntax in here as well for posterity? :) I saw you mention in Slack that we can set a default tab, and that doesn't seem to be covered by the K8s reference page

@RichieEscarez  said he'll take a look at this, so I'll assign him as reviewer.

**Comment by mattmoor** [comment]
```
{{< tabs name="serving_networking" default="Istio" >}}
{{% tab name="Contour" %}}
Stuff
{{< /tab >}}

{{% tab name="Istio" %}}
More Stuff
{{< /tab >}}
{{< /tabs >}}
```

**Comment by samodell** [comment]
image

^From Slack #docs channel

**Comment by RichieEscarez** [comment]
Awesome Matt! (sorry i cant dive in here with you but maybe this helps) I'm parsing the if statement stuff. My initial guess is that its there for the case were someone defines multiple {{% tab ...... shortcodes in the same file and it checks first to see if a "tab title" or `name` already exists in the "array" (`slice`).

Maybe instead of just `asdf` we use something helpful like "TAB TITLE DOES NOT EXIST" (or something to that affect to indicate how to fix it).

One other important item would be to make sure that our [readfile](https://github.com/knative/website/blob/master/layouts/shortcodes/readfile.md) shortcode [(see implemented examples)](https://github.com/knative/docs/search?q=readfile&unscoped_q=readfile) works within these tabs. (that would be the ideal situation so that we can just include other file content in a tab and avoid duplicating and hardcoding copies of the same stuff everywhere).

Lastly, we should add examples of how to use the tabs on the [smoketest page](https://knative.dev/smoketest/)

ie. Does this work:

```
{{< tabs name="serving_networking" default="Istio" >}}
{{% tab name="Contour" %}}
Stuff
{{< /tab >}}

{{% tab name="README" %}}
{{% readfile file="README.md" %}}
{{< /tab >}}

{{% tab name="CODE SAMPLE" %}}
{{< readfile file="/community/samples/serving/helloworld-java-quarkus/service.yaml" code="true" lang="yaml" >}}
{{< /tab >}}


{{% tab %}}
Does this error our or show "asdf" in the tab title?
{{< /tab >}}
{{< /tabs >}}
```

**Comment by samodell** [comment]
@mattmoor  I'm happy add these tabs to the smoketest page (it's over in the Docs repo): https://github.com/knative/docs/blob/bb824f1f97c11dc5c1998c9f623b68c94fda0707/smoketest.md

**Comment by mattmoor** [comment]
I update the message

**Comment by mattmoor** [comment]
@RichieEscarez README works, CODE SAMPLE doesn't, I don't even see a 4th tab (or errors in the console).

**Comment by RichieEscarez** [comment]
ok cool. thats good news. code sample might have failed due to the file path (ie. its DNE locally, etc) but as long as we can pull in content from another file, thats good enough for me. LGTM from me (ill let Sam handle the merging)

**Pull request merged**